packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,420 @@
+      /**
+       * Id to pass to input
+       */
+      inputId: string;
+      setInputId: (inputId: string) => void;
+      /**
+       * Ref for input subcomponent
+       */
+      inputRef: React.RefObject<HTMLInputElement>;
+      /**
+       * Ref for open button subcomponent
+       */
+      openButtonRef: React.RefObject<HTMLButtonElement>;
+      /**
+       * Callback for closing expandable searchbox
+       */
+      onCollapse?: () => void;
+      /**
+       * Callback for expanding searchbox
+       */
+      onExpand?: () => void;
+      /**
+       * SearchBox state
+       */
+      isExpanded?: boolean;
+      /**
+       * Is SearchBox expandable
+       */
+      expandable?: boolean;
+    }
+  | undefined
+>(undefined);
+
+// ----------------------------------------------------------------------------
+
+type SearchBoxOwnProps = {
+  /**
+   * Whether the searchbox is expandable.
+   * @default false
+   */
+  expandable?: boolean;
+  /**
+   * SearchBox state toggle.
+   */
+  isExpanded?: boolean;
+  /**
+   * Function that is called on expanding searchbox.
+   */
+  onExpand?: () => void;
+  /**
+   * Function that is called on collapsing searchbox.
+   */
+  onCollapse?: () => void;
+  /**
+   * Input props when using default searchbox.
+   */
+  inputProps?: React.ComponentPropsWithoutRef<'input'>;
+  /**
+   * Modify size of the searchbox and it's subcomponents.
+   */
+  size?: 'small' | 'large';
+};
+
+const SearchBoxComponent = React.forwardRef((props, ref) => {
+  useTheme();
+
+  const {
+    size,
+    expandable = false,
+    isDisabled = false,
+    onCollapse: onCollapseProp,
+    onExpand: onExpandProp,
+    isExpanded: isExpandedProp,
+    children,
+    inputProps,
+    className,
+    ...rest
+  } = props;
+
+  const uid = useId();
+  const [inputId, setInputId] = React.useState(uid);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const openButtonRef = React.useRef<HTMLButtonElement>(null);
+
+  const [localExpanded, setLocalExpanded] = React.useState(isExpandedProp);
+
+  const isExpanded = isExpandedProp ?? localExpanded;
+
+  const onCollapse = () => {
+    setLocalExpanded(false);
+    onCollapseProp?.();
+    queueMicrotask(() => openButtonRef.current?.focus({ preventScroll: true }));
+  };
+
+  const onExpand = () => {
+    setLocalExpanded(true);
+    onExpandProp?.();
+    queueMicrotask(() => inputRef.current?.focus({ preventScroll: true }));
+  };
+
+  return (
+    <SearchBoxContext.Provider
+      value={{
+        size,
+        isDisabled,
+        onCollapse,
+        onExpand,
+        inputRef,
+        inputId,
+        setInputId,
+        openButtonRef,
+        isExpanded,
+        expandable,
+      }}
+    >
+      <InputFlexContainer
+        ref={ref}
+        className={cx({ 'iui-expandable-searchbox': expandable }, className)}
+        data-iui-size={size}
+        isDisabled={isDisabled}
+        data-iui-expanded={isExpanded}
+        aria-label='SearchBox'
```

Comment:
@mayank99 What do you think about adding this here?

Reply:
I guess just make it say "Search" for now. But in our documentation, we will need to clearly mention all the places we are setting aria-label

Reply:
This is applied on the container, which is not right. The container is a generic `<div>`, so adding `aria-label` to it does nothing.

Reply:
It announces "SearchBox, group" on the first visit.
Without aria-label here:
https://user-images.githubusercontent.com/83585998/236845633-149629c6-8af8-4b93-b4bb-653da0c53c81.mov

With aria-label here:
https://user-images.githubusercontent.com/83585998/236845744-2915f9c4-965e-4e2e-ab53-d743535aa0e2.mov

Reply:
i don't think it will work reliably in all cases and it is also not recommended. https://www.htmhell.dev/adventcalendar/2022/7/#non-interactive-elements

if you want to have an aria-label, try also adding `role=region`; or remove it.

Reply:
Removed it. We can always add it later.